### PR TITLE
Support: auto-run swimlane converter per case in scene_test

### PR DESCRIPTION
**Pull request by ChaoWao**
## Summary

Wire `tools/swimlane_converter.py` into both pytest and standalone scene test paths so `--enable-profiling` runs always produce a usable merged swimlane without a manual follow-up step. Also revamps the scene test CLI: `--all-cases` -> `--manual {exclude,include,only}`, and `--case` now accepts compound `ClassA::Foo` selectors.

## Per-case conversion flow

For each case that runs with profiling enabled:

1. Snapshot `outputs/perf_swimlane_*.json` and CANN device log dir
2. Run the case
3. Wait briefly (<=2s) for the new perf file
4. **Rename** the perf file to embed `Class_Case` in its name — the runtime writes `perf_swimlane_<second-timestamp>.json`, so two cases completing within the same second would otherwise clobber each other
5. Wait (<=15s, hardware only) for the new CANN device log
6. Invoke the converter with `--device-log` when available; the output name is derived from the renamed input stem by the converter's existing `_resolve_output_path` logic

### `--manual` replaces `--all-cases`

The old flag was misnamed; its actual semantics were already "include manual cases":

```
--manual exclude   # default, skip manual cases
--manual include   # run manual cases alongside normal ones
--manual only      # run only manual cases (new capability)
```

### `--case` is now repeatable and supports compound forms

```
--case Foo              # any class has a case named Foo
--case ClassA::Foo      # qualified: only ClassA::Foo
--case ClassA::         # all cases in ClassA
--case A::x --case B::y # repeatable
```

`--case` and `--manual` compose orthogonally. A fully-qualified selector that resolves only to manual cases while `--manual=exclude` raises with a hint to pass `--manual include`.

Standalone `run_module` strictly validates selectors (unknown class/case or empty selection → error). pytest path silently skips non-matches to match `-k` ergonomics.

## Other changes

- `st_worker` fixture stashes allocated device ids on the Worker so `test_run` can discover the primary device for log snapshots — fixes a bug where multi-class tests sharing a `DevicePool` range all snapshotted the first `--device` only
- `run_module` warns when `--enable-profiling` is combined with a hardware platform but no explicit `-d/--device`
- `docs/testing.md` documents the new selector and manual model

## Test plan

- [x] Helper unit checks (`_parse_case_selector`, `_match_selectors`, `_select_cases`) — edge cases: compound forms, manual override, unknown class/case errors
- [x] `python -m py_compile` on touched files
- [x] Pre-commit hooks (ruff, pyright, markdownlint) pass
- [ ] Live run: `pytest examples/a2a3/tensormap_and_ringbuffer/vector_example/test_vector_example.py --platform a2a3sim --enable-profiling --case '::'` verifies per-case swimlane generation
- [ ] Live run on hardware: `python test_*.py -p a2a3 -d 0 --enable-profiling` verifies device-log pairing and rename path

### CODE REVIEW

File: simpler_setup/scene_test.py [hunk 1]
```diff
@@ -335,6 +337,201 @@ def _resolve_chip_entry_paths(entry, cls_dir):
...
+        return (cls_part or None, case_part or None)
+    return (None, value)
+
+
+def _match_selectors(cls_name: str, case_name: str, selectors: list[tuple]) -> bool:
+    """True if ``(cls_name, case_name)`` matches any selector (empty list means no selector filter)."""
+    for sel_cls, sel_case in selectors:
+        if (sel_cls is None or sel_cls == cls_name) and (sel_case is None or sel_case == case_name):
+            return True
+    return False
```
[the remark this answers is not in the document]
Comment (ChaoWao) [reply simpler_setup/scene_test.py:361]: Fixed in `b137607` — `_match_selectors` now returns `True` for empty selectors, matching the docstring. Call-site guards `if selectors and ...` in `_select_cases` and `test_run` simplified to `if not _match_selectors(...)`.

File: simpler_setup/scene_test.py [hunk 2]
```diff
@@ -335,6 +337,201 @@ def _resolve_chip_entry_paths(entry, cls_dir):
...
+        cmd += ["-d", str(device_id)]
+    try:
+        result = subprocess.run(cmd, check=True, capture_output=True, text=True)
+        if result.stdout:
+            logger.info(result.stdout)
+        logger.info("Swimlane JSON generation completed")
+    except subprocess.CalledProcessError as e:
+        logger.warning(f"Failed to generate swimlane JSON: {e}")
+        if e.stderr:
+            logger.debug(f"stderr: {e.stderr}")
```
[the remark this answers is not in the document]
Comment (ChaoWao) [reply simpler_setup/scene_test.py:498]: Fixed in `b137607` — `e.stdout` is now also logged at debug level on `CalledProcessError`.
